AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthUserAttributeKey+Extension.swift

```diff
@@ -8,75 +8,89 @@
+    private static let pictureRawValue = "picture"
+    private static let preferredUsernameRawValue = "preferred_username"
+
+    public init(rawValue: String) {
+        switch rawValue {
+        case AuthUserAttributeKey.addressRawValue:
+            self = .address
+        case AuthUserAttributeKey.birthDateRawValue:
+            self = .birthDate
+        case AuthUserAttributeKey.emailRawValue:
+            self = .email
+        case AuthUserAttributeKey.familyNameRawValue:
+            self = .familyName
+        case AuthUserAttributeKey.genderRawValue:
+            self = .gender
+        case AuthUserAttributeKey.givenNameRawValue:
+            self = .givenName
+        case AuthUserAttributeKey.localeRawValue:
+            self = .locale
+        case AuthUserAttributeKey.middleNameRawValue:
+            self = .middleName
+        case AuthUserAttributeKey.nameRawValue:
+            self = .name
+        case AuthUserAttributeKey.nicknameRawValue:
+            self = .nickname
+        case AuthUserAttributeKey.phoneNumberRawValue:
+            self = .phoneNumber
+        case AuthUserAttributeKey.pictureRawValue:
+            self = .picture
+        case AuthUserAttributeKey.preferredUsernameRawValue:
+            self = .preferredUsername
+        default:
+            self = .custom(rawValue)
+        }
+    }
+
+    public var rawValue: String {
         switch self {
         case .address:
-            return "address"
+            return AuthUserAttributeKey.addressRawValue
         case .birthDate:
-            return "birthdate"
+            return AuthUserAttributeKey.birthDateRawValue
         case .email:
-            return "email"
+            return AuthUserAttributeKey.emailRawValue
         case .familyName:
-            return "family_name"
+            return AuthUserAttributeKey.familyNameRawValue
         case .gender:
-            return "gender"
+            return AuthUserAttributeKey.genderRawValue
         case .givenName:
-            return "given_name"
+            return AuthUserAttributeKey.givenNameRawValue
         case .locale:
-            return "locale"
+            return AuthUserAttributeKey.localeRawValue
         case .middleName:
-            return "middle_name"
+            return AuthUserAttributeKey.middleNameRawValue
         case .name:
-            return "name"
+            return AuthUserAttributeKey.nameRawValue
         case .nickname:
-            return "nickname"
+            return AuthUserAttributeKey.nicknameRawValue
         case .phoneNumber:
-            return "phone_number"
+            return AuthUserAttributeKey.phoneNumberRawValue
         case .picture:
-            return "picture"
+            return AuthUserAttributeKey.pictureRawValue
         case .preferredUsername:
-            return "preferred_username"
+            return AuthUserAttributeKey.preferredUsernameRawValue
         case .custom(let attribute):
             return "custom: \(attribute)"
```

Comment:
i think this should just be `return attribute`, since what was passed in was the `attribute` string and stored in `.custom` and then converting it back should be lossy

Reply:
Changed the logic to return what was passed in during init. Custom attribute require to be init with "custom:" as the prefix. Here are some examples:
```
let x = AuthUserAttributeKey(rawValue: "someUnknownattribute")
x.rawValue // returns "someUnknownattribute"
x == .unknown("someUnknownattribute")

let x = AuthUserAttributeKey(rawValue: "custom:someattribute")
x.rawValue // returns "custom:someattribute"
x == .custom("someattribute")

let x = AuthUserAttributeKey(rawValue: "middle_name")
x.rawValue // returns "middle_name"
x == .middleName
```

This way the logic doesnot break if the service adds new attributes. It will just fall back to unknown enum meaning that the attribute is not yet known to the Amplify.Auth.